lib/src/install.rs

```diff
@@ -522,6 +604,24 @@ impl FromStr for MountSpec {
     }
 }
 
+impl InstallToDiskOpts {
+    pub(crate) fn validate(&self) -> Result<()> {
+        if !self.composefs_native {
+            // Reject using --boot without --composefs
+            if self.composefs_opts.boot != BootType::default() {
+                anyhow::bail!("--boot must not be provided without --composefs");
+            }
+        }
+
+        // Can't add kargs to UKI
```

Comment:
This is complicated but I think actually what we'll need to do at some point is ensure that when e.g. `ukify` is run that we participate in assembling the kernel commandline - and that part should *consume* (i.e. delete) the `/usr/lib/bootc/kargs.d` directory.